### PR TITLE
EVG-7387: use correct home directory for legacy SSH

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-7387

This is a compatibility fix for Windows legacy and non-legacy provisioning.

For almost all cases, paths to Windows files and directories have to be specified as native file paths when passing them as command line arguments (i.e. if you use `/`, it refers to `C:/` not `C:/cygwin`). Jasper processes (which run via exec), are consistent with this behavior because all file paths have to be in native Windows style (e.g. `C:/cygwin/home/Administrator/evergreen.exe fetch --dir C:/cygwin/data`). However, when invoking executables directly through legacy SSH, we have to use Unix-style paths (e.g. `/home/Administrator/evergreen.exe`) since their paths are evaluated directly in Cygwin bash. The Unix-style path rule doesn't apply for CLI flags on the evergreen binary, which are native Windows-style paths.

We have some dependency in legacy provisioning on the fact that `/data` is symlinked at the root of the C drive in order to do `evergreen fetch`.

The best solution is probably to require `RootDir` be specified for both legacy and non-legacy hosts on Windows so that we can use native Windows paths correctly. There's a ticket for follow-on work to do this: https://jira.mongodb.org/browse/EVG-7439

tl;dr - Jasper is consistent and will always require native WIndows paths for everything (binaries, files, directories). Cygwin is painfully inconsistent since depends on the context in which the filepath is being used.

### CODE REVIEW

File: model/distro/distro.go [hunk 1]
```diff
@@ -388,8 +388,12 @@ func (d *Distro) ExecutableSubPath() string {
 	return filepath.Join(arch, d.BinaryName())
 }
 
-// HomeDir gets the absolute path to the home directory for this distro's user.
+// HomeDir gets the absolute path to the home directory for this distro's user
+// for non-legacy provisioned hosts.
 func (d *Distro) HomeDir() string {
+	if d.LegacyBootstrap() {
```
Comment (tychoish) [inline model/distro/distro.go]: why not invert these cases?

Comment (Kimchelly) [reply model/distro/distro.go]: I don't get what you mean. Legacy SSH is the exceptional edge case that has to be written exactly like this and depends on being run in a shell - all other provisioning methods can't rely on running in a shell, and therefore actually specify a path.

Anyways, it doesn't really matter because I'm going to remove this piece of code soon anyways.
